### DISCUSSION TITLE
infer结果和predict结果不一样！

### DISCUSSION THREAD

**Discussion by justajustin** [PaddlePaddle/PaddleOCR#12576] [Q&A]
Infer的处理和predict的处理不同，导致两个代码结果差别很大。我下载PPOCR-v4训练模型，在infer_rec上都是乱码，predict_rec上效果正常。版本是release-2.7。

**Comment by metoogo** [comment]
你应该是忘记设置字典了，找一下字典文件在哪里然后配置一下就可以了，我的是在这里，配置一下这个configs参数：character_dict_path: ppocr/utils/ppocr_keys_v1.txt

**Comment by justajustin** [comment]
> 你应该是忘记设置字典了，找一下字典文件在哪里然后配置一下就可以了，我的是在这里，配置一下这个configs参数：character_dict_path: ppocr/utils/ppocr_keys_v1.txt

我已经设置了，对比了infer和predict的文件都是一样的字典

**Comment by metoogo** [comment]
使用的是同一个模型吗？检测的模型跟识别的模型是否都指向同一个

**Comment by justajustin** [comment]
> 使用的是同一个模型吗？检测的模型跟识别的模型是否都指向同一个

不是同一个，用的分别是ppocr-v4_server的训练模型和推理模型

[a comment by an automated account is not shown]

**Comment by SWHL** [comment]
如果问题仍然存在，建议拉取最新main分支代码，再次尝试
